.github/scripts/platform_lib.py

```diff
@@ -0,0 +1,377 @@
+import abc
+
+import sys
+import boto3
+import os
+from enum import Enum
+from fabric.api import *
+from ci_variables import local_fsim_dir, ci_azure_sub_id
+import datetime
+import pytz
+
+from azure.mgmt.resource import ResourceManagementClient
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.compute import ComputeManagementClient
+import azure.mgmt.resourcegraph as arg
+
+from typing import Any, Callable, Dict, List
+
+# Reuse manager utilities
+# Note: ci_firesim_dir must not be used here because the persistent clone my not be initialized yet.
+sys.path.append(local_fsim_dir + "/deploy")
+from awstools.awstools import get_instances_with_filter
+
+# This tag is common to all instances launched as part of a given workflow
+unique_tag_key = 'ci_workflow_id'
+# Managers additionally have this empty tag defined
+manager_filter = {'Name': 'tag:ci_manager', 'Values' : ['']}
+
+# The following is an enum used to represent the different platforms
+class Platform(Enum):
+    ALL = 'all'
+    AWS = 'aws'
+    AZURE = 'azure'
+    def __str__(self) -> str:
+        return self.value
+
+def get_platform_enum(platform_string: str) -> Platform:
+    if platform_string == 'aws':
+        return Platform.AWS
+    elif platform_string == 'azure':
+        return Platform.AZURE
+    elif platform_string == 'all':
+        return Platform.ALL
+    else:
+        raise Exception(f"Invalid platform string: '{platform_string}'")
+
+class PlatformLib(metaclass=abc.ABCMeta):
+    """
+    This is a class hierarchy to support multiple platforms in FireSim CI
+    Note that there are 2 terms used to refer to machines running CI: Managers and Instances
+
+    Managers - refer to a manager instance, this only really matters for AWS, but it's the main machine
+    running CI
+
+    Instances - a more general term, but can refer to runfarm / buildfarm instances run by CI as well
+    """
+
+    @abc.abstractmethod
+    def get_filter(self, tag_value: str) -> Dict:
```

Comment:
Similar to the other method with `tag_value`, I think it should be renamed to `workflow_id`

Reply:
Im sensing this trend with all the `tag_value` arg. names. Is there a case where the `tag_value` would not be the workflow ID?